ci/ray_ci/pipeline/gap_filling_scheduler.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime, timedelta
+from typing import List, Dict, Optional, Any
+
+from pybuildkite.buildkite import Buildkite
+
+from release.ray_release.aws import get_secret_token
+from release.ray_release.configs.global_config import AWS_SECRET_BUILDKITE
+
+
+class GapFillingScheduler:
+    """
+    This buildkite pipeline scheduler is responsible for scheduling gap filling builds
+    when the latest build is failing.
+    """
+
+    buildkite = None
+
+    def __init__(
+        self,
+        buildkite_organization: str,
+        buildkite_pipeline: str,
+    ):
+        self.buildkite_organization = buildkite_organization
+        self.buildkite_pipeline = buildkite_pipeline
+        GapFillingScheduler._init_buildkite()
+
+    @classmethod
+    def _init_buildkite(cls):
+        if not cls.buildkite:
+            cls.buildkite = Buildkite()
+            cls.buildkite.set_access_token(get_secret_token(AWS_SECRET_BUILDKITE))
```

Comment:
i want to avoid loading secret token multiple times, but i guess it's not not too bad; the plan is to move the secret token to the global_config (because it is defined once per repository, not the GappFillingScheduler class)

Reply:
you can always send in the config as a arg or parameter. global configs should not be imported into sub modules, but should be dispatched into sub modules on init process.

otherwise, you would have a config file ever growing, with business logic from all modules, all mixed together, and have a dependency structure that every logic depends on every other logic's config structs.

Reply:
sound good, i'll add the global configs only on the top level script and pass down only the required values here